src/main/java/io/debezium/connector/yugabytedb/YugabyteDBChangeRecordEmitter.java

```diff
@@ -196,8 +196,7 @@ private Object[] columnValues(List<ReplicationMessage.Column> columns, TableId t
             undeliveredToastableColumns.remove(columnName);
             int position = getPosition(columnName, table, values);
             if (position != -1) {
-                Object value = column.getValue(() -> (BaseConnection) connection.connection(),
-                        connectorConfig.includeUnknownDatatypes());
```

Comment:
why is this change needed,  connection.connection() can return null/connection depending on the context

Reply:
The `YugabyteDBConnection` object doesn't have any context of what type of data it will be used to decode. In case an array type is encountered, then a JDBC connection is created to be used otherwise it would be just the `YugabyteDBConnection` object.